src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs

```diff
@@ -263,4 +278,8 @@ private static bool IsTls13Disabled(bool isServer)
 #endif
         return false;
     }
+
+    private static bool IsAppLocalMsQuicAllowed() => AppContextSwitchHelper.GetBooleanConfig(
+        "System.Net.Quic.AppLocalMsQuic",
+        "DOTNET_SYSTEM_NET_QUIC_APPLOCALMSQUIC");
```

Comment:
I do not think it makes sense to make this configurable via env variable. If somebody sets this env variable, all of their .NET apps (e.g. including .NET SDK) are going to break - except the one app that has the quic ..dll next to it.

(I know that local libICU has the env variable, but I do not think it is a good pattern to follow.)

Reply:
I have not noticed that you have a fallback to the system QUIC when the local one fails to load. Is it the desired behavior with the switch?

(For local ICU, there are no fallbacks. If the app asked for local ICU and the local ICU failed to load, it is going to fail.)